aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -735,4 +786,29 @@ private Set<Tag> toTags(Map<String, String> properties, String prefix) {
   public Map<String, String> s3BucketToAccessPointMapping() {
     return s3BucketToAccessPointMapping;
   }
+
+  public <T extends S3ClientBuilder> void applyS3ServiceConfigurations(T builder) {
+    builder
+        .dualstackEnabled(s3DualStackEnabled)
+        .serviceConfiguration(
+            S3Configuration.builder()
+                .pathStyleAccessEnabled(s3PathStyleAccess)
+                .useArnRegionEnabled(s3UseArnRegionEnabled)
+                .accelerateModeEnabled(s3AccelerationEnabled)
+                .build());
+  }
+
+  public SdkHttpClient.Builder configureHttpClientBuilder() {
```

Comment:
can we follow the same pattern as the method above, like `public <T extends XXX> void configureHttpClientBuilder(T builder)` to set the HTTP client? XXX is the common parent class of all AWS client builders

Reply:
I changed the method to this pattern by integrating the `.httpClientBuilder` into this function. I also renamed it as `applyHttpClientConfiguration` since now it will be used with `applyMutation` like the method above.